src/niftyone/figures/generator.py

```diff
@@ -104,22 +136,44 @@ def _figure_name(self) -> None:
             self.entities["extra_entities"]["figure"] = self.view_kwargs["figure"]
             del self.view_kwargs["figure"]
 
-    def generate(self, record: pd.Series, out_dir: Path, overwrite: bool) -> None:
+    def generate(
+        self,
+        records: list[pd.Series],
+        out_dir: Path,
+        overwrite: bool,
+    ) -> None:
         """Main call for generating view."""
         if not self.view_fn:
             raise ValueError("View is not provided, unable to create generator.")
 
         # Update figure name if necessary
         self._figure_name()
 
-        img_path = Path(record["finfo"]["file_path"])
+        img_path = Path(records[0]["finfo"]["file_path"])
         logging.info("Processing: %s", img_path)
+
         img = nib.nifti1.load(img_path)
         img = noimg.to_iso_ras(img)
 
-        existing_entities = BIDSEntities.from_dict(record["ent"])
+        # Handle overlays - currently only handles 1, update to handle multiple
+        overlays = []
+        # Temporary logic to handle multiple overlays
+        if len(records) > 2:
+            raise NotImplementedError("Multi-image overlay not yet implemented")
+
+        if len(records) > 1:
+            for overlay_record in records[1:]:
+                overlay_path = Path(overlay_record["finfo"]["file_path"])
+                overlay = nib.nifti1.load(overlay_path)
+                overlays.append(noimg.to_iso_ras(overlay))
+                self.view_kwargs["overlay"] = overlays.pop()
+
+        existing_entities = BIDSEntities.from_dict(records[0]["ent"])
         out_path = existing_entities.with_update(self.entities).to_path(prefix=out_dir)
         out_path.parent.mkdir(exist_ok=True, parents=True)
+
         if not out_path.exists() or overwrite:
             logging.info("Generating %s", out_path)
             self.view_fn(img, out_path, **self.view_kwargs)
+
+        plt.close("all")
```

Comment:
Yeah, I thought about that (if I am understanding what you mean here), but I think in some cases we do return the figure. I don't recall what happens if the figure gets closed before it is used on a return. This also ensures that all figures end up getting closed (in case it wasn't for some reason).